app/controllers/items_controller.rb

```diff
@@ -62,10 +62,38 @@ def safe_params
   end
 
   def find_items
-    if current_user
-      @items = Item.where(user: current_user)
+    sort = params[:sort]
```

Comment:
not 100% sure what you mean - my intention is to avoid getting the full array and having to manipulate it based on the sort params, but rather use the sort params to pull the precise array needed. Are you saying it would be easier to run an if/else for the current_user first - then do case statements to sort for each outcome?

Reply:
Why don't you look for a gem to do this for you? I can image you might also want pagination, which is handle with limit and offset